apps/xen-tg-app/src/components/Capabilities.tsx

```diff
@@ -61,16 +71,58 @@ const mutationFn = async ({
 }
 
 const Capabilities = () => {
-  const queryClient = useQueryClient()
-
-  const { data: capabilities } = useQuery<{ items: TAgent[]; total: number }>({
+  const sentinelRef = useRef<HTMLDivElement | null>(null)
+  const observerRef = useRef<IntersectionObserver | null>(null)
+  const {
+    data: capabilities,
+    // error,
+    fetchNextPage,
+    hasNextPage,
+    isFetching,
+    isFetchingNextPage,
+    // status,
+  } = useInfiniteQuery({
     queryKey: ['capabilities'],
-    queryFn: queryFn('getCapabilities', {
+    queryFn: queryFn('getCapabilities'),
+    initialPageParam: {
       offset: 0,
       limit: 10,
-    }),
+    },
+    getNextPageParam: (lastPage, __, lastPageParam) => {
+      if (!lastPage?.length) {
```

Comment:
I suspect there is an extra query here. You can use the `total' parameter from the server response to avoid this.
```json
{
    "jsonrpc": "2.0",
    "result": {
        "items": [{},{},{},{},{}],
        "total": 412 // <-----
    },
    "id": 4
}
```

Reply:
fixed